### ISSUE TITLE
[FEATURE] Anchor Display to Frame/WeakAura

### ISSUE THREAD

**Issue by dmora** [Hekili/hekili#1199]
### Before You Begin

- [X] I confirm that I have downloaded the latest version of the addon.
- [X] I checked for an [existing, open ticket](https://github.com/Hekili/hekili/labels/enhancement) for this request and was not able to find one.
- [X] I edited the title of this feature request (above) so that it describes the issue I am reporting.

### Feature Request

as a user that has many weakauras, custom UIs, i would like to have the option to anchor a display to a specific frame/WeakAura so the positioning of the display is always relative to that frame. This will help when a dynamic group changes positions and one has to go manually and update the position of the Display in the addon.

### Contact Information

SoulBits#1792

**Comment by Hekili** [comment]
I'll leave this open as an idea, and I've considered it before (especially anchoring to the personal nameplate), but it's weird to me to think people want to look around for guidance from the addon like that.

**Comment by dmora** [comment]
Well, i usually look down a bit to check the cooldowns in the middle screan WAs, so i have a placeholder in the middle to show the recommended ability, but the placeholder can change position depending on number of cooldowns or special abilities according to classes and specs. So i always have to re-position it.

Other way to go around this is to allow positioning per class/specs.

**Comment by TimeCrisis86** [comment]
I'd like to second this request, specifically being able to anchor to the Name Plate. I have a WA that lets me anchor my pally holy power counter to my target's Elvui nameplate, allowing me to keep my eyes on the target. Would love to have the Hekili addon also be anchored to the name plate. Thanks!

**Comment by S1lverone** [comment]
Did we find a solution to anchor it to the nameplate?

here is a code for anchoring to nameplate.

function()
    local region = aura_env.region
    local plate = C_NamePlate.GetNamePlateForUnit("target")
    if plate then
        region:ClearAllPoints()
        region:SetPoint("TOP", plate, "TOP", 0, 20)
        region:Show()
    else
        region:Hide()
    end
    return true
end

**Comment by Hekili** [comment]
It's not simply a matter of anchoring to a nameplate.  I'm still considering this.  Nameplates hide and show and the addon has to also know how to recover gracefully under those circumstances.  Not a top priority, but I'll mark this as backlogged for now.

**Comment by mmathias01** [comment]
For anyone still interested in this I had a similar idea and rather than change the way that Hekili works I decided to try and do it as a WeakAura. 

Here is one that anchors to the mouse cursor: https://wago.io/tpLAPJ9xC

The version that is posted as of me writing this works pretty well and is really usable, but will still throw some lua errors in certain edge cases and has some missing features. I have a new version I will be posting pending the outcome of a pull request I submitted that helps to close some of those gaps.
